spring-rabbit-stream/src/main/java/org/springframework/rabbit/stream/producer/RabbitStreamTemplate.java

```diff
@@ -188,19 +224,49 @@ public CompletableFuture<Boolean> convertAndSend(Object message, @Nullable Messa
 	@Override
 	public CompletableFuture<Boolean> send(com.rabbitmq.stream.Message message) {
 		CompletableFuture<Boolean> future = new CompletableFuture<>();
-		createOrGetProducer().send(message, handleConfirm(future));
+		observeSend(message, future);
 		return future;
 	}
 
+	private void observeSend(com.rabbitmq.stream.Message message, CompletableFuture<Boolean> future) {
+		Observation observation = RabbitStreamTemplateObservation.TEMPLATE_OBSERVATION.observation(
+				this.observationConvention, DefaultRabbitStreamTemplateObservationConvention.INSTANCE,
+				() -> new RabbitStreamMessageSenderContext(message, this.beanName, this.streamName),
+					obtainObservationRegistry());
+		observation.start();
+		try {
+			createOrGetProducer().send(message, handleConfirm(future, observation));
+		}
+		catch (Exception ex) {
+			observation.error(ex);
+			observation.stop();
+			future.completeExceptionally(ex);
+		}
+	}
+
+	@Nullable
+	private ObservationRegistry obtainObservationRegistry() {
+		if (!this.observationRegistryObtained && this.observationEnabled && this.applicationContext != null) {
+			if (this.applicationContext != null) {
```

Comment:
I guess we don't need the first one then.
Even if there is no ctx, we still need to set `observationRegistryObtained` to avoid extra steps into this `if` for subsequent calls.